### ISSUE TITLE
Line Stacks showing wrong values in axisPointer tooltip

### ISSUE THREAD

**Issue by lucasgcb** [apache/echarts#15435]
### Version
5.1.2

### Reproduction link
[https://echarts.apache.org/examples/en/editor.html?c=doc-example/line-stack-tiled](https://echarts.apache.org/examples/en/editor.html?c=doc-example/line-stack-tiled)

### Steps to reproduce
In the example provided, simply hit the top right "Stack" button to activate the stack.
Then, hover over the stacked lines and check the inconsistency in Y axis.

In any other situation, 
Create stacked lines, describe a tooltip with axis trigger.
Then, hover over the stacked lines and check the inconsistency in Y axis.

### What is expected?
Values in the axisPointer tooltip to be consistent to what is being shown in the plot.

### What is actually happening?
Values retaining their original implementation format, leading to visual-value discrepancies.
![plot](https://i.imgur.com/92mKUh8.png)

---
This is a problem that can be seen in the very documentation.

In my situation, I'm producing confidence bands using stacks and knowing the values for upper and lower bounds are interesting for the user.

However, the implementation values are shown rather than what is being presented in the plot. This makes the use of tooltips in stacked lines less useful, and even damaging since viewers may interpret a different value. What is shown and what is calculated  can lead to disinformation.

Either the documentation should be updated, or there should be an option to match what is shown in the plot without front-end interference through formatter shenanigans (which could lead to even more discrepancies).

**Comment by susiwen8** [comment]
We can't not change the origin value, that is the natural of tooltip

**Comment by lucasgcb** [comment]
> We can't not change the origin value, that is the natural of tooltip

What? 

Yes, you can change the value. I have changed it manually with a formatter after personally noticing it, however this is terrible practice because it means that *the tooltip is wrong by default*. This is not explicit in the documentation. It could be fixed without even breaking the API.
 
This behavior is not natural of tooltips; tooltips are meant to inform the user about meaningful elements. The implementation value for line stacks are not meaningful and in the context of the plot; the tooltip is absolutely incorrect when line stacking is used and could lead to terrible mistakes.

**Comment by susiwen8** [comment]
For example, if data were 
```js
['tree', 30],
['building', 40],
['street', 2],
```
and tooltip shows for stacked line
```js
['tree', 30],
['building', 70],
['street', 72],
```

Then user might think the number of `street` is 72, but the true number is 2.

ECharts respect input data value. User can change it by `formatter`, but that is user intention, not ECharts'.

**Comment by lucasgcb** [comment]
> For example, if data were
> 
> ```js
> ['tree', 30],
> ['building', 40],
> ['street', 2],
> ```
> 
> and tooltip shows for stacked line
> 
> ```js
> ['tree', 30],
> ['building', 70],
> ['street', 72],
> ```
> 
> Then user might think the number of `street` is 72, but the true number is 2.
> 
> ECharts respect input data value. User can change it by `formatter`, but that is user intention, not ECharts'.

In your example, 2 is indeed the true number for the _street_ series. However, when [line stack](https://echarts.apache.org/en/option.html#series-line.stack) is enabled, the _street_ series is never actually shown in the plot.

If the data is inputted as a stack, it's not longer about _street_. It's _a new series_ on the plot, with tree+building+street. The true number for this is not 2. The true number is 30 + 40 + 2. Otherwise, it would not be a stack. The tooltip will completely ignore this and simply show "street" with the same color identity, even if it no longer is in the plot.

On the very plot I attached:

![image](https://user-images.githubusercontent.com/7332704/127755658-46b8f5ac-1537-4874-9d4f-929c66b24afa.png)

The yellow series in this plot is a stack. Its "true number" is not 234, because that's not what is shown in the plot; what is shown in the plot is a _new series with stacked values_ while retaining the color identity of the `意向` series. The plot does not reflect this whatsoever, and becomes disinformation.

With the current way the tooltip is implemented, there's no way to tell when stack is enabled nor is there an easy way to fix the tooltip other than messing with the formatter, which can then lead to more issues. This discrepancy in more subtle line series can lead to terrible results.

**Comment by pissang** [comment]
@lucasgcb `stack` doesn't mean it will transform the data into a sum value. It's a form of line/area chart displaying that can show a different aspect of data. For example, we have a dataset that about the sales of three products A, B, C. Without stack mode, we can see the trends of each product individually and compares them. And with stack mode, we can see the trends of the sum of all these three products and the distribution of each product. But we can't let product C display the sum value of all three products in the tooltip in the stacked mode. It's wrong.

Perhaps the example misleads you. If you wan't to display the sum value in one series. Just calculate the data before you give them to echarts.

**Comment by lucasgcb** [comment]
> @lucasgcb `stack` doesn't mean it will transform the data into a sum value. It's a form of line/area chart displaying that can show a different aspect of data.
> 
> For example, we have a dataset that about the sales of three products A, B, C. Without stack mode, we can see the trends of each product individually and compares them. But with stack mode, we can see the trends of the total sale of all these three products and the distribution of each product. But we can't let product C display the sum value of all three products in the tooltip in the stacked mode. It's wrong.
>  If you wan't to display the sum value in one series, just calculate the data before you give them to echarts

Unfortunately the problem in that example is that the viewer would _already_ be seeing Product C displaying the sum value when lines get stacked. The graph would calculate it as so, the label would have "Product C", and the color would be the same. The tooltip information will never match the plot when this option is used.

It's already shown wrongly on the plot. That's the problem; the stack is not "Product C", but resulting data of A + B + C. 
The labels/tooltip should reflect this.

And no, I do not wish for the sum value. This is a bug report. I solved my problem before creating this issue; I have followed the [confidence band example](https://echarts.apache.org/examples/en/editor.html?c=confidence-band) and went down a rabbit hole trying to understand this tooltip discrepancy. Then, I double checked the documentation and realized that the tooltip implementation indeed has this strange behavior for this option which I then worked around of. I reported it here, because it's a problem that's easy to miss and could create chaos.

This should have, at the very least, a warning firing off when tooltips are enabled alongside stacked series without a formatter; it's that bad.
